lib/_stream_pump.js

```diff
@@ -150,14 +150,14 @@ function destroyer(stream, reading, writing, _callback) {
       return stream.abort();
       // request.destroy just do .end - .abort is what we want
 
-    if (isFn(stream.destroy)) return stream.destroy();
+    if (isFn(stream.destroy)) return stream.destroy(err);
```

Comment:
fyi, this wont work on older streams, which is why it was not calling it with `err` before